module3/lessons/Week4/UpdatingResources.md

```diff
@@ -0,0 +1,219 @@
+    var html = await response.Content.ReadAsStringAsync();
+
+    // Assert
+    response.EnsureSuccessStatusCode();
+    Assert.Contains("Goofy", html);
+    Assert.Contains("Documentary", html);
+    Assert.DoesNotContain("Comedy", html);
+}
+```
+
+</section>
+
+### the Order of Our Tests
+
+Before we implement our controller action, we need to address one aspect of testing - the order and timing of each of our tests.  By default, xUnit will run our tests randomly, and as quickly as possible (some tests even run at the same time!).  Generally, this is a good thing - we want our tests to be individually robust, and not rely on any _other_ test to run successfully. In this case, though, we need to control our test runs a bit more.
+
+If left to their default settings, we could have two tests running at the same time (concurrently) that would put conflicting data into our test database.  xUnit provides a simple way to group tests that need to be run strictly one at a time.  We will use an attribute to group all of our Controller tests into one collection:
+
+```c#
+// MoviesControllerTests.cs
+
+namespace MvcMovie.FeatureTests
+{
+    [Collection("Controller Tests")]
+    public class MovieControllerTests : IClassFixture<WebApplicationFactory<Program>>
+    {
+        ...
+```
+
+Any test class in this collection will be run one at a time, reducing the chance for a database conflict.
+
+<section class='call-to-action' markdown='1'>
+
+**With a Partner**  
+Implement the controller action that we need to update records in our database.
+
+</section>
+
+<section class='answer' markdown='1'>
+
+### One Solution
+```c#
+[HttpPost]
+[Route("/Movies/{id:int}")]
+public IActionResult Update(Movie movie)
+{
+    _context.Movies.Update(movie);
+    _context.SaveChanges();
+
+    return RedirectToAction("show", new { id = movie.Id });
+}
+```
+
+</section>
+
+## TDD Updating related resource
+**Code-Along**  
+As a class, we are going to write tests, and implement code to satisfy these user-stories:
+
+```
+As a User
+When I visit a Movie Show page
+Then I see links to edit each review
+```
+
+```
+As a User
+When I visit 'Movies/1/Reviews/1/edit'
+Then I see the name of the Movie and,
+Then I see a form to edit the review
+    And the review fields are pre-populated with their current value
+```
+
+```
+As a User
+When I Submit the review-edit form
+Then I am redirected to the Movie Show Page
+    And I see my updated review
+```
+
+
+<section class='instructor-notes' markdown='1'>
+
+Completed code can be found on the update-resources branch of the [MvcMovieStarter](https://github.com/turingschool-examples/MvcMovieStarter)
+
+</section>
+
+
+## Checks for Understanding
+* Our path to get the 'edit' form includes the resource id.  Why do we _not_ need an id in our path to get the 'new' form?
+* What are the similarities between creating a new record, and editing an existing one?
```

Comment:
These checks for understanding are great! Love that they are deeper level questions. Might be helpful to also think about the differences between creating and editing?

Reply:
Yes, love that distinction!